service/matching/config.go

```diff
@@ -148,6 +148,56 @@ type (
 
 		// Retry policy for fetching user data from root partition. Should retry forever.
 		GetUserDataRetryPolicy backoff.RetryPolicy
+
+		loadCause loadCause
+	}
+
+	loadCause   int
+	unloadCause int
+)
+
+const (
+	loadCauseUnspecified loadCause = iota
+	loadCauseTask
+	loadCauseQuery
+	loadCauseDescribe
+	loadCauseUserData
+	loadCauseNexusTask
+	loadCausePoll
+	loadCauseOther
+)
+
+const (
+	unloadCauseUnspecified unloadCause = iota
+	unloadCauseInitError
+	unloadCauseIdle
+	unloadCauseMembership
+	unloadCauseConflict
+	unloadCauseShuttingDown
+	unloadCauseForce
+	unloadCauseOther
+)
+
+var (
+	loadCauseToString = map[loadCause]string{
```

Comment:
You can add a `.String` method to both of these types that use these so that go will auto-format them using `%s`

Reply:
oh, yeah, that'd be simpler. I was thinking I didn't want to be the first use of [stringer](https://pkg.go.dev/golang.org/x/tools/cmd/stringer) in the codebase but that's also an option

Reply:
I just went for stringer. How important is it that those generated files (or any of our generated files really) have a copyright header? I could fix it but it's annoying since the tool doesn't do it itself